### PR TITLE
(RE-7326) Remove jira class and functionality from packaging

### PR DESCRIPTION
**Pull request by McdonaldSeanp**
We are moving ticket generation automation out of packaging, so remove all the
methods and classes we use for that (as well as the tasks)

We are moving these jira tasks to https://github.com/puppetlabs/winston

### CODE REVIEW

**Comment by McdonaldSeanp** [comment]
Blocked on https://github.com/puppetlabs/winston/pull/1

File: Gemfile [hunk 1]
```diff
@@ -8,17 +8,3 @@ group :development, :test do
...
-  # was introduced. I mean, we could probably constrain this in general
-  # because ActiveSupport is not a thing we want to rely on, but
-  # being conservative is probably safest.
-  #
-  # - Ryan McKern, 2016-01-19
-  if RUBY_REVISION < 50295
-    gem "activesupport", "< 5.0.0", require: false
-  end
-  gem "jira-ruby"
-end
```
Comment (mckern) [inline Gemfile]: YAAAAAS

File: lib/packaging/util.rb [hunk 2]
```diff
@@ -11,7 +11,6 @@ module Pkg::Util
   require 'packaging/util/git'
   require 'packaging/util/gpg'
   require 'packaging/util/jenkins'
-  require 'packaging/util/jira'
```
Comment (mckern) [inline lib/packaging/util.rb]: ##### YAAAAAS #####

File: lib/packaging/util/jira.rb [hunk 3]
```diff
@@ -1,167 +0,0 @@
...
-        transition = issue.transitions.build
-        transition.save!("transition" => { "id" => options_hash[:status] })
-      end
-
-      return issue.key, issue.id
-    rescue Exception => e
-      fail "Cannot create Jira Ticket with fields #{fields}"
-    end
-  end
-end
```
Comment (mckern) [inline lib/packaging/util/jira.rb]: #### YAAAAAS ####

**Comment by shrug** [comment]
We need to make sure the other teams that use these are on board. Client team for sure, not sure who else uses?

**Comment by mckern** [comment]
its happening

**Comment by McdonaldSeanp** [comment]
@shrug that's cool, how do we find out who uses it?

**Comment by shrug** [comment]
ping @whopper - any thoughts about who else we should ping for this?

**Comment by whopper** [comment]
@shrug for client, I think this is fine as long as the same functionality exists elsewhere and there's some way for us to figure out how to use it.

Also note that there's also the puppet-agent release ticket task: `tasks/puppet-agent-release-tickets.rake`. It doesn't look like this PR removes it, but it probably should?

**Comment by shrug** [comment]
@whopper @barlowst @kylog yep it's all going into it's own repo:  https://github.com/puppetlabs/winston

**Comment by McdonaldSeanp** [comment]
@whopper yeah, I missed those ones. Fixed now

**Comment by shrug** [comment]
👍

**Pull request merged**